### ISSUE TITLE
[GlobalStep] PHP Fatal error appears after updating the WooPayment plugin to 8.5.0-test-2 version.

### ISSUE THREAD

**Issue by gglobalstep** [Automattic/woocommerce-payments#9676]
### Describe the bug
PHP Fatal error appears after updating the WooPayment plugin to 8.5.0-test-2 version.

### Environment:
Woocommerce Version: WooCommerce  9.4.0-rc.2
WooCommerce Payments: 8.5.0-test-2
WordPress version: v6.7-RC2

PC:
Windows 10,
Chrome(Version 130.0.6723.92)
Firefox (Version 132.0.1)

### Steps To Reproduce:
1. Create any test site.
2. Install and activate all the required plugins.
3. Install and activate the Woocommerce version 9.4.0-rc.2 plugins.
4. Install the latest stable version of WooCommerce Payments via WP-ADMIN -> Plugins page.
5. Complete WooCommerce Payments KYC flow.
6. Go to WP-ADMIN -> Plugins & click on Upload Plugin to upload the testing package(i.e.8.5.0-test-2 version).
7. Observe that, PHP Fatal error appears after updating the WooPayment plugin to 8.5.0-test-2 version.

### Instruction Link:
https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#plugin-update-via-plugins-page

### Actual behavior
PHP Fatal error appears after updating the WooPayment plugin to 8.5.0-test-2 version.

### Expected behavior
WooPayment plugin(8.5.0-test-2 version) should be updated successfully without any errors.

### Screenshots
![Image](https://github.com/user-attachments/assets/d2808852-12cb-46f6-8d19-537230ca1268)

*Isolating the problem (mark completed items with an [x]):*
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.4.0-rc.2
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.2
Log Directory Writable: ✔
WP Version: 6.7-RC2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.1.30
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 8.10.1
OpenSSL/3.0.14

SUHOSIN Installed: –
MySQL Version: 10.11.9-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 1
page: 7
post: 2
revision: 1
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (6) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.3
Code Snippets: by Code Snippets Pro – 3.6.5.1
Query Monitor: by John Blackbourn – 3.16.4
WooPayments: by WooCommerce – 8.5.0-test-2
WooCommerce: by Automattic – 9.4.0-rc.2
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.6.1

### Inactive Plugins (0) ###


### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
db.php: Query Monitor Database Class (Drop-in)
object-cache.php: Memcached

### Must Use Plugins (1) ###

load.php: by  –

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 300 KB

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/ -  Contains the woocommerce/cart block
Checkout: #9 - /checkout/ -  Contains the woocommerce/checkout block
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.2
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/error.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/success.php


### WooPayments ###

Version: 8.5.0-test-2
Connected to WPCOM: Yes
WPCOM Blog ID: 238714765
Account ID: acct_1QHm4uCSAHSKSD8L
Payment Gateway: Enabled
Test Mode: Enabled
Enabled APMs: card
WooPay: Enabled (product,cart,checkout)
WooPay Incompatible Extensions: No
Apple Pay / Google Pay: Enabled (product,cart,checkout)
Fraud Protection Level: basic
Multi-currency: Enabled
Auth and Capture: Enabled
Documents: Disabled
Logging: Enabled

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings
use-wp-horizon

Daily Cron: ✔ Next scheduled: 2024-11-06 12:39:23 +00:00
Options: ✔
Notes: 58
Onboarding: skipped

### Action Scheduler ###

Complete: 5
Oldest: 2024-11-05 12:39:43 +0000
Newest: 2024-11-05 12:51:18 +0000

Pending: 1
Oldest: 2024-11-06 12:39:43 +0000
Newest: 2024-11-06 12:39:43 +0000


### Status report information ###

Generated at: 2024-11-05 14:01:40 +00:00
```
</details>

**Comment by haszari** [comment]
I see `multi-currency/Settings.php` in the fatal error, so adding multicurrency focus area. Since this is a fatal I'm marking as critical. Possibly this is an artifact of the testing setup, either way we need to resolve quickly (by fixing or confirming won't affect merchants).

**Comment by haszari** [comment]
Assigning to @ricardo based on an internal (slack) discussion. Feel free to reassign as appropriate.

**Comment by nagpai** [comment]
I did a quick test yesterday, on existing stable versions of WC 9.3.3 and WordPress 6.6.2. and was able to replicate the error. 
Hence, we can say this is not a regression introduced because of the RC versions of either.

It appears only if we use the query monitor, hence I missed it while upgrading our Apple Pay test site, as a part of the release lead workflow.

**Comment by ricardo** [comment]
Changed the priority as this is not reproducible in production. It's a problem currently only in `develop`. Should be fixed by #9681.
